theano/tensor/opt.py

```diff
@@ -5304,7 +5308,9 @@ def local_zero_div(node):
     if isinstance(node.op, T.Elemwise) and isinstance(
             node.op.scalar_op, (theano.scalar.IntDiv, theano.scalar.TrueDiv)):
         if local_mul_canonizer.get_constant(node.inputs[0]) == 0:
-            return [broadcast_like(0, node.outputs[0], node.fgraph)]
+            ret = broadcast_like(0, node.outputs[0], node.fgraph)
+            ret.tag.values_eq_approx = values_eq_approx_remove_nan
```

Comment:
I agree for that one: it is replacing `0/x` by `0`, so when evaluated at `x == 0`, a NaN is removed.
I think fixing this one should be enough to fix the test, and the ones in `local_merge_switch_same_cond`, and probably in `local_fill_sink`, should be removed.